samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/PublishAssessmentListener.java

```diff
@@ -285,25 +306,73 @@ private void publish(AssessmentFacade assessment,
        // this  might change later
         log.warn(gbe.getMessage(), gbe);
         // Add a global message (not bound to any component) to the faces context indicating the failure
-        String err=(String)ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.AuthorMessages",
-                                                 "gradebook_exception_min_points");
-        FacesContext.getCurrentInstance().addMessage(null, new FacesMessage(err));
+        String errorMessage = ContextUtil.getLocalizedString(
+                SamigoConstants.AUTHOR_BUNDLE, "gradebook_exception_min_points");
+        FacesContext.getCurrentInstance().addMessage(null, new FacesMessage(errorMessage));
         throw new AbortProcessingException(gbe);
     } catch (InvalidGradeItemNameException gbe) {
         log.warn(gbe.getMessage(), gbe);
-        String err=(String)ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.AuthorMessages",
-                                                 "gradebook_exception_title_invalid");
-        FacesContext.getCurrentInstance().addMessage(null, new FacesMessage(err));
+        String errorMessage = ContextUtil.getLocalizedString(
+                SamigoConstants.AUTHOR_BUNDLE, "gradebook_exception_title_invalid");
+        FacesContext.getCurrentInstance().addMessage(null, new FacesMessage(errorMessage));
         throw new AbortProcessingException(gbe);
     } catch (Exception e) {
         log.warn(e.getMessage(), e);
         // Add a global message (not bound to any component) to the faces context indicating the failure
-        String err=(String)ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.AuthorMessages",
-                                                 "gradebook_exception_error");
-        FacesContext.getCurrentInstance().addMessage(null, new FacesMessage(err));
+        String errorMessage = ContextUtil.getLocalizedString(
+                SamigoConstants.AUTHOR_BUNDLE, "gradebook_exception_error");
+        FacesContext.getCurrentInstance().addMessage(null, new FacesMessage(errorMessage));
         throw new AbortProcessingException(e);
     }
 
+    // Safe Exam Browser upload
```

Comment:
this code feels like it should not be here..... it should be in the SEB code

Reply:
I am not sure how this could be in the SEB code, since this is not part of the delivery, but part of the settings. When an assessment is published and has an uploaded SEB config file, I need to insert the published assessment URL there and save the resource in the metadata. It would be nice if the SecureDeliveryModule spec could interact with the publishing process. But I think extending the interface would cause more trouble then this. What do you think?

Reply:
check out the SecureDeliveryProctorio code. My questions would be:

a) Do you have to save the resource in the metadata?
b) Can you do the SEB work just-in-time (e.g., when the user wants to take the assessment) instead of doing it when the assessment is being saved?
c) Extending the interface with one or two methods would be much better than inserting product-specific code here.

Reply:
Thank you very much for the feedback Sam!

Reply:
Thanks a lot Sam!